cli/js/testing.ts

```diff
@@ -159,32 +158,34 @@ class TestApi {
       tests: this.testsToRun.length
     };
 
+    const results: TestResult[] = [];
     const suiteStart = +new Date();
-    for (const testCase of this.testCases) {
-      const { fn, result } = testCase;
-      let shouldBreak = false;
-      try {
-        const start = +new Date();
-        await fn();
-        result.duration = +new Date() - start;
-        result.passed = true;
-        this.stats.passed++;
-      } catch (err) {
-        result.passed = false;
-        result.error = err;
-        this.stats.failed++;
-        shouldBreak = this.failFast;
-      } finally {
-        result.hasRun = true;
-        yield { kind: TestEvent.Result, result };
-        if (shouldBreak) {
-          break;
+    for (const { name, fn, skip } of this.testsToRun) {
+      const result: Partial<TestResult> = { name };
+      if (skip) {
+        result.status = TestStatus.Skipped;
+        this.stats.ignored++;
+      } else {
+        try {
+          const start = +new Date();
+          await fn();
+          result.duration = +new Date() - start;
```

Comment:
please capture test duration even if it errors

Reply:
It's not there for `skip = true`, or does Rust do it?